Gems/EMotionFX/Code/EMotionFX/Source/AnimGraphInstance.cpp

```diff
@@ -283,28 +280,27 @@ namespace EMotionFX
     {
         // check how many parameters we need to add
         const ValueParameterVector& valueParameters = mAnimGraph->RecursivelyGetValueParameters();
-        const int32 numToAdd = static_cast<uint32>(valueParameters.size()) - mParamValues.GetLength();
+        const ptrdiff_t numToAdd = aznumeric_cast<ptrdiff_t>(valueParameters.size()) - mParamValues.size();
         if (numToAdd <= 0)
         {
             return;
         }
 
         // make sure we have the right space pre-allocated
-        mParamValues.Reserve(static_cast<uint32>(valueParameters.size()));
+        mParamValues.reserve(valueParameters.size());
 
         // add the remaining parameters
-        const uint32 startIndex = mParamValues.GetLength();
-        for (int32 i = 0; i < numToAdd; ++i)
+        const size_t startIndex = mParamValues.size();
+        for (ptrdiff_t i = 0; i < numToAdd; ++i)
```

Comment:
Why ptrdiff_t? Do we have more instances like that?

Reply:
It is a `ptrdiff_t` because that is the signed version of a `size_t`.